### PR TITLE
Liza & Amanda Labyrinth! 

### PR DESCRIPTION
**Pull request by amandaflod**
Link to our deployed project:

https://infallible-pasteur-82e0d5.netlify.app/

My reflections on how this week's project turned out:

Great teamwork and a lot of learnings! 🐯

### CODE REVIEW

File: code/src/reducers/gameInfo.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export const generateDescription = () => {
+  //const URL_START_GAME = "https://wk16-backend.herokuapp.com/start"
+
+  return (dispatch) => {
+    fetch('https://wk16-backend.herokuapp.com/start',{
+      method:'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({'username':'TestUser'})
+    }
```
Comment (Katsiez) [inline code/src/reducers/gameInfo.js]: Perhaps I have missed it, but does the randomized username show anywhere?

File: code/src/components/Buttons.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const actions = useSelector(store => store.gameDetails.gameDetails.actions)
+  const userName = useSelector(store => store.gameDetails.userName)
+  
+  const handleOnClick = (action) => {
+  dispatch(getMoves(action,userName));
+  }
+
+  return (
+    <>
+      {actions.map(item => {
```
Comment (Katsiez) [inline code/src/components/Buttons.js]: Okay, mapping the buttons-how clever 😍

File: code/src/reducers/gameInfo.js [hunk 3]
```diff
@@ -31,7 +35,12 @@ export const getMoves = (action) => {
...
-      .then(data => 
-        dispatch(gameDetails.actions.setMoves(data)));
+      .then(data => {
+        if (data.actions.length === 0) {
+          dispatch(gameDetails.actions.gameFinished())
+        }
+        dispatch(gameDetails.actions.setMoves(data));
+    });
       };
 };
```
Comment (Katsiez) [inline code/src/reducers/gameInfo.js:39]: Interesting to see other ways of implementing the 'game over' function, especially with an if statement and a length property! We have done it a bit different, by clearing the local storage-nonetheless, this is once again smart and simple. ☺️

**Review by Katsiez — COMMENTED** [review]
Dear Amanda and Liza, great job on the Labyrinth project!✨🌈🧩 I love how clean your design is, the fact that you made a reusable button component, and how easy your code is to comprehend. I love that you included a clean loader and a simpler way of re-initializing the game upon completion. Like any project, even this one could of course be expanded, if you ever wish to do so. A simple yet effective thing I can recommend implementing would be the local storage/persisted state in Redux, so that the game progress is not lost on refresh. This is only up to you, of course! Great team work!👯‍♀️

File: code/src/components/GameBegin.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        <>
+          <div> Welcome to the game!  </div>
+          <StyledButton onClick={onGameBegin}>{">"} play</StyledButton>
+        </>
+      ) : (
+          <>
+            <Text>{gameData.description}</Text>
+            {gameData.description && <Buttons />}
+            {!gameOff && <StyledButton onClick={handleHistoryBack}
+              disabled={historyContainer === 1} style={{ color: 'red' }}>{"<"} go back</StyledButton>}
```
Comment (lindahz) [inline code/src/components/GameBegin.js:48]: Is the disabled function working? For example, when I am at coordinates 0.0 the button should be disabled, but I can see the focus outline when I click on the button, even if I cannot go back. We actually have the same issue, not sure how to solve it. 😩 Maybe the code coaches knows?

**Review by lindahz — APPROVED** [review]
Yey 🥰, great job Amanda and Liza! 

We really love your text based design, it feels like you’re directly in the terminal playing the game. So cool. The design also looks great with mobile and tablet. We can also see that you used minimalistic code to accomplish the design. ⭐️

The app and its functionalities works great. 🤩 The code is also well written and understandable, especially the reducers with comments on each state. 

Again, GREAT job, we're so impressed. 💛

File: code/src/lib/StyledButton.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+
+import styled from 'styled-components'
+
+export const StyledButton = styled.button`
+  color: #38F237;
+  font-family: ‘Inconsolata’, monospace;
+  background: transparent;
+  background: none;
```
Comment (EmOlsz) [inline code/src/lib/StyledButton.js:8]: Ups, duplicated `background` property

File: code/src/reducers/gameDetails.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  reducers: {
+    //this action sets the description in the beginning of the game
+    setDescription: (state, action) => {
+      state.gameDetails = action.payload;
+    },
+    //this action sets the following moves and takes care of the history 
+    setMoves: (state, action) => {
+      if (state.gameDetails) {
+        state.history = [...state.history, state.gameDetails]
```
Comment (EmOlsz) [inline code/src/reducers/gameDetails.js:21]: Great job, that's a clever idea!

File: code/src/reducers/gameInfo.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import { gameDetails } from './gameDetails';
+
+export const getDescription = (userName) => {
+  return (dispatch, getState) => {
+    dispatch(gameDetails.actions.setLoading(true));
+    fetch('https://wk16-backend.herokuapp.com/start', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ username: userName })
```
Comment (EmOlsz) [inline code/src/reducers/gameInfo.js:9]: If you would name your argument in line 3 as `username` (without camelCase), then we could simplify this line of code :

```
body: JSON.stringify({ username })
```

**Review by EmOlsz — COMMENTED** [review]
Congratulations Liza & Amanda, good job :) I like your design and logic behind code, so :

```
<Congratulations
  textMessage="Project approved!"
/>
```

**Pull request closed without merging**